templates/entry/_options.html.twig

```diff
@@ -1,136 +1,244 @@
-            </a>
-        </li>
-        <li>
-            <a href="{{ options_url('sortBy', 'active') }}"
-               class="{{ html_classes({'active': route_has_param('sortBy', 'active')}) }}">
-                {{ 'active'|trans }}
-            </a>
-        </li>
-        <li>
-            <a href="{{ options_url('sortBy', 'commented') }}"
-               class="{{ html_classes({'active': route_has_param('sortBy', 'commented')}) }}">
-                {{ 'commented'|trans }}
-            </a>
+    <menu class="options__filter">
+        <li class="dropdown">
+            <button aria-label="{{ 'sort_by'|trans }}"
+                    title="{{ 'sort_by'|trans }}"><i
+                        class="fa-solid fa-sort" aria-hidden="true"></i>
+                <span>{{ criteria.getOption('sort')|trans }}</span>
+            </button>
+            <ul class="dropdown__menu">
+                <li>
+                    <a href="{{ options_url('sortBy', 'top') }}"
+                    class="{{ html_classes({'active': criteria.getOption('sort') == 'top'}) }}">
+                        {{ 'top'|trans }}
+                    </a>
+                </li>
+                <li>
+                    <a href="{{ options_url('sortBy', 'hot') }}"
+                    class="{{ html_classes({'active': criteria.getOption('sort') == 'hot'}) }}">
+                        {{ 'hot'|trans }}
+                    </a>
+                </li>
+                <li>
+                    <a href="{{ options_url('sortBy', 'newest') }}"
+                    class="{{ html_classes({'active': criteria.getOption('sort') == 'newest'}) }}">
+                        {{ 'newest'|trans }}
+                    </a>
+                </li>
+                <li>
+                    <a href="{{ options_url('sortBy', 'active') }}"
+                    class="{{ html_classes({'active': criteria.getOption('sort') == 'active'}) }}">
+                        {{ 'active'|trans }}
+                    </a>
+                </li>
+                <li>
+                    <a href="{{ options_url('sortBy', 'commented') }}"
+                    class="{{ html_classes({'active': criteria.getOption('sort') == 'commented'}) }}">
+                        {{ 'commented'|trans }}
+                    </a>
+                </li>
+            </ul>
         </li>
-    </menu>
-    <menu class="options__view">
         <li class="dropdown">
             <button aria-label="{{ 'filter_by_time'|trans }}"
-                    title="{{ 'filter_by_time'|trans }}"><i
-                        class="fa-solid fa-clock" aria-hidden="true"></i> <span>{{ get_time_param_translated() }}</span>
+                    title="{{ 'filter_by_time'|trans }}">
+                <i aria-hidden="true" class="fa-solid fa-clock"></i>
+                {% if showFilterLabels == 'on' or (showFilterLabels == 'auto' and criteria.getOption('time') != '∞') %}
+                    <span>{{ criteria.getOption('time')|trans }}</span>
```

Comment:
I'm fine if you want to ditch `get_time_param_translated` completely including inside the menu selection, and delete the code for it. My intent with it was to sanitize the url paramters being sent to the translate package, but perhaps symfony does that already and it was overly cautious of me

However, a slight nitpick but I think it would be better to show `All` rather than `∞` to be consistent. @nobodyatroot and I actually discussed removing `∞` altogether as it's a bit out of place, but even if it's how the routing uses, I think the text `All` might be more clear

Reply:
Yeah, absolutely on both counts. I think in the long run we should be eliminating get_time_param_translated() in addition to a bunch of other similar one-off functions, and just using Criteria.getOption() as the unified way. My thinking was to leave a lot of them in place temporarily, convert the filtering display to the new way one controller at a time, and then clean up and remove the no-longer-needed one-offs at the end.

And yes, I actually tried to remove `∞` but backed it up once I ran into a fair hornet's nest of stuff. I think in the long run, it'd be good to do a whole cleanup pass and remove a lot of the inconsistency in how strings are used to represent these options values internally -- `∞` versus `all` is far from the worst of it; for a single option of the subscription-status filter we use `featured`, `fav`, `favorite`, `favourite`, or `favourites` at different places in the code, and then attempt to translate among those values at various points so that code that expects one version of the string won't break when it gets the wrong version. Honestly, it's madness. I had to sort of draw the line of what level of cleanup it made sense to do while I was doing this already-too-invasive change, but if you're open to the idea I'd like to go in at some point later and take a pass at making it all consistent internally. It would have saved me a certain amount of time and confusion on this change if it had been consistent going into it.

Changing the displayed value to `all` is easy if that's preferable though, sure, I can do that.

Reply:
Latest change changes the display value to "All".